src/cascade/executor/no_covariate_main.py

```diff
@@ -158,95 +169,57 @@ def age_year_from_data(df):
         values.sort()
         results[topic] = pd.DataFrame({topic: values})
         LOGGER.debug(f"{topic}: {values}")
-    return results["age"], results["year"]
-
-
-def integrand_outputs(rates, location_id, age, time):
-    """
-    The internal model declares what outputs it wants.
-    """
-    age_time = np.array(list(it.product(age["age"].values, time["year"].values)), dtype=np.float)
-    entries = list()
-    for rate in (RATE_TO_INTEGRAND.get(r) for r in rates):
-        entries.append(pd.DataFrame({
-            "integrand": rate,
-            "location_id": location_id,  # Uses location_id, not node id
-            "weight": "constant",  # Assumes a constant rate exists.
-            # Uses modeler versions of age and year, not age and time.
-            "age_start": age_time[:, 0],
-            "age_end": age_time[:, 0],
-            "year_start": age_time[:, 1],
-            "year_end": age_time[:, 1]
-        }))
-    return pd.concat(entries, ignore_index=True)
-
-
-def build_smoothing_grid(age, time):
-    """Builds a default smoothing grid with uniform priors."""
-    age_time = np.array(list(it.product(age["age"].values, time["year"].values)), dtype=np.float)
-    return pd.DataFrame({
-        "age": age_time[:, 0],
-        "year": age_time[:, 1],
-        "value_prior": "uniform01",
-        "age_difference_prior": "uniform",
-        "time_difference_prior": "uniform",
-        "const_value": np.NaN,
-    })
+    return AgeTimeGrid(results["age"].age, results["year"].year)
 
 
 def build_constraint(constraint):
     """
     This makes a smoothing grid where the mean value is set to a given
     set of values.
     """
-    return pd.DataFrame({
-        "age": constraint["age_start"],
-        "year": constraint["year_start"],
-        "value_prior": None,
-        "age_difference_prior": "uniform",
-        "time_difference_prior": "uniform",
-        "const_value": constraint["mean"],
-    })
-
-
-def internal_model(model_context, inputs):
-    config = model_context.parameters
-    model = model_context.input_data
+    ages = constraint["age_start"].unique().tolist()
+    times = constraint["year_start"].unique().tolist()
+    grid = AgeTimeGrid(ages, times)
+    smoothing_prior = PriorGrid(grid)
+    smoothing_prior[:, :].prior = UniformPrior(-np.inf, np.inf, 0)
```

Comment:
This prior with the infinities... does it work for Dismod? It's a completely uninformative prior. That's what the are called. Difficult to use.

Reply:
This is functionally identical to what you had in your code, unless I misunderstood what you were doing which is possible. Whether it makes sense or not, I defer to you.